src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -4168,4 +4238,93 @@ struct C
         var comp = CreateCompilationWithRequiredMembers(code);
         CompileAndVerify(comp).VerifyDiagnostics();
     }
+
+    [Fact]
+    public void HasExistingObsoleteAttribute_RequiredMembersOnSelf()
+    {
+        var code = """
+            using System;
+            class C
+            {
+                public required int Prop { get; set; }
+
+                [Obsolete("Reason 1", false)]
+                public C() { }
+
+
+                [Obsolete("Reason 2", true)]
+                public C(int unused) { }
+            }
+            """;
+
+        var comp = CreateCompilationWithRequiredMembers(code);
+        CompileAndVerify(comp, symbolValidator: validate, sourceSymbolValidator: validate).VerifyDiagnostics();
+
+        static void validate(ModuleSymbol m)
+        {
+            var c = m.GlobalNamespace.GetTypeMember("C");
+            var ctors = c.GetMembers(".ctor").Cast<MethodSymbol>().ToArray();
+
+            Assert.Equal(2, ctors.Length);
+
+            assertAttributeData(ctors[0], "Reason 1", expectedError: false);
+            assertAttributeData(ctors[1], "Reason 2", expectedError: true);
+
+            static void assertAttributeData(MethodSymbol ctor, string expectedReason, bool expectedError)
```

Comment:
nit: consider flattening inner local function instead of nesting

Reply:
I prefer nesting if the local function isn't used by the surrounding context.